src/main/java/com/pjohnson_wtc/bank_account/Admin.java

```diff
@@ -0,0 +1,138 @@
+package com.pjohnson_wtc.bank_account;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class Admin {
+	
+	private static Logger logger = Logger.getLogger("com.pjohnson_wtc.bank_account.admin");
+	
+	//List of all accounts
+	private List<Account> accounts = null;
+	
+	//Method for creating accounts from csv file
+	public List<Long> createAccount(String filename) {
+		
+		//Call readFromCsv() to read csv file and format into List of Lists
+		List<List<String>> accountApplicants = readFromCsv(this.getClass().getClassLoader().getResourceAsStream(filename));
+		
+		//Initialize accounts to a new ArrayList
+		accounts = new ArrayList<Account>();
+		
+		//Create list for storing and returning account numbers
+		List<Long> accountNums = new ArrayList<Long>();
+		
+		//Loop through list of account applicants
+		for (List<String> applicant : accountApplicants) {
+			
+			//Skip to next line if validation fails
+			if (!validateApplicant(applicant)) continue;
+
+			//Declare local variable for saving new Account objects
+			Account newAccount = finallyCreateAccount(applicant);
+			
+			//Add new Account to Admin.accounts and account number list
+			accounts.add(newAccount);
+			accountNums.add(newAccount.getAccountNumber());
+		}
+		
+		logger.log(Level.INFO, accountNums.size() + " accounts created");
+		
+		//Return list of account numbers
+		return accountNums;
+	}
+	
+	//Method for reading data from csv file
+	private List<List<String>> readFromCsv(InputStream filepath) {
+		
+		//Create new ArrayList of Lists of Strings for storing formatted data
+		List<List<String>> accountApplicants = new ArrayList<List<String>>();
+		
+		//Read through csv file, line by line
+		try (BufferedReader br = new BufferedReader(new InputStreamReader(filepath))) {
+			
+		//Store current line in local variable
+		String line;
+		
+		//For each line, split the String at each comma and store the results in an array of Strings
+		while ((line = br.readLine()) != null) {
+			
+			String[] values = line.split(",");
+			
+			//Convert this array of Strings to a List and to accountApplicants
+			accountApplicants.add(Arrays.asList(values));
+		}
+		
+		//Return formatted list of data
+		return accountApplicants;
+		
+		} catch (IOException ioe) {
+			
+			logger.log(Level.WARNING, "Could not read file", ioe.toString());
+			return null;
+		}
+	}
+	
+	//Method for validating input from csv
+	private boolean validateApplicant(List<String> applicant) {
+		
+		//Return false if account type is neither Savings nor Checking
+		if (!(applicant.get(2).equals("Savings") || applicant.get(2).equals("Checking"))) {
```

Comment:
Null check missing. It's a run time exception and should be handled

Reply:
Good point - will do